tests/generic/assert_evm_address.sql

```diff
@@ -0,0 +1,14 @@
+{% test assert_evm_address(model, column_name) %}
+
+    with unit_test as
+    (
+        select
+            rlike({{ column_name }}, '0[x][0-9a-f]{40}') as assert_evm_address
+        from {{ model }} m
+        where blockchain in ('ethereum', 'optimism', 'arbitrum', 'avalanche_c', 'polygon', 'bnb', 'gnosis', 'fantom') -- TODO: turn into generic macro (all_evm_chains fails atm)
```

Comment:
We can remove this "where" clause and make use of the generic `where` config for tests to filter any model. 
docs for more info: https://docs.getdbt.com/reference/resource-configs/where#examples


```
tests:
    - assert_evm_address:
        where: "blockchain in ( 'ethereum, 'optimism, ...)"
```

Reply:
That should make this generic test applicable to any models! :partying_face:

Reply:
oh interesting!

I applied the filter at the definition level because the test looks specifically for evm address format, and will fail on valid e.g. solana addresses.

were you thinking  of that scenario? or scenarios where we want to use this in a single chain model (e.g. `blur/ethereum`) when we apply it upstream?

btw moved the test to #2995